### DISCUSSION TITLE
Ways to read mcap with topic and time filtering

### DISCUSSION THREAD

**Discussion by wy-moonind** [foxglove/mcap#1086] [General]
Hello, i was trying to read a MCAP file which is recorded based on ros2 messages. The current way is to use rosbag2_cpp/reader/sequential_reader to read every single messages inside the bag based on sequential time. This way does not accept any time or topic filtering, the only way is to use some if-else to filter the topic and time, which is very inefficient. 

Is there any api which i can read only the selected time and topics directly from the mcap file?

 If so, i would be very appreciate for some examples... Big help!

**Comment by defunctzombie** [comment]
If you are using ROS and the ROS apis I recommend posting this request to the specific ROS repo that contains those reader APIs.

If you are using the mcap c++ library from the mcap repo the `readMessages` method of the reader accepts options. One of those options is a topic filter: https://mcap.dev/docs/cpp/r37F558A23E8E6347

**Reply by deo-abhijit** [reply]
Hi! Thanks for the amazing tool! 

I was wondering if there is a way to do same using python?

**Reply by deo-abhijit** [reply]
sorry, found this: 

https://mcap.dev/docs/python/mcap-apidoc/mcap.reader#mcap.reader.McapReader
